api/users/views.py

```diff
@@ -401,8 +405,48 @@ def get_default_odm_query(self):
             Q('is_registration', 'eq', True) &
             Q('contributors', 'eq', user._id)
         )
+
         permission_query = Q('is_public', 'eq', True)
         if not current_user.is_anonymous():
             permission_query = (permission_query | Q('contributors', 'eq', current_user._id))
         query = query & permission_query
         return query
+
+class UserNodeLogs(JSONAPIBaseView, generics.ListCreateAPIView, ODMFilterMixin, UserMixin):
+
+    serializer_class = NodeLogSerializer
+    pagination_class = UserNodeLogPagination
+    view_category = 'users'
+    view_name = 'node_logs'
+
+    ordering = ('-date', )
+
+    def get_default_odm_query(self):
+        user = self.get_user()
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        raw_nodes = Node.find(
+            Q('contributors', 'eq', user) &
+            Q('is_folder', 'ne', True) &
+            Q('is_deleted', 'ne', True)
+        )
+        nodes = [each for each in raw_nodes if each.is_public or each.can_view(auth)]
+        if not nodes:
+            raise NotFound
+        query = [Q('params.node', 'eq', node._id) for node in nodes]
+        return reduce(lambda x, y: x | y, query)
```

Comment:
Sure! So I'm using ` reduce ` which takes in a function with two arguments, and a list. It takes in the first two elements of the list and applies the function (x is first element, y is second), then the output of that function (in this case x | y) becomes the first argument of the function again, and the second argument is replaced by the third element, and so on until it has gone through the entire list. So example:
```
l  = [ 1, 2, 3, 4]

print reduce(lambda x, y: x+y, l)  #  10
# so it does ((1+2) + 3) + 4)
```
A lambda is just an unnamed function, so my thing above ( ` lambda x, y: x | y `), is equivalent to:
```
def func(a, b):
    return a | b
```
does that clear it up a little bit? Let me know if it doesn't, or if that is not what you were asking about :P